pennylane/workflow/construct_batch.py

```diff
@@ -81,8 +84,108 @@ def _get_full_transform_program(
     return program + qnode.device.preprocess_transforms(config)
 
 
+def _validate_level(
+    level: Optional[Literal["top", "user", "device", "gradient"] | int | slice],
+) -> None:
+    """Check that the level specification is valid.
+
+    Args:
+        level: The level specification from user input
+
+    Raises:
+        ValueError: If the level is not recognized
+    """
+    if level is None or isinstance(level, int) or isinstance(level, slice):
```

Comment:
```suggestion
    if level is None or isinstance(level, (int, slice)):
```
to shut up codefactor. sorry about that suggestion.